### PR TITLE
Add preference controlled toggling between full or truncated file path in top bar - #13987-

### PR DESCRIPTION
**Pull request by nyteksf**
Issue #13987

### CODE REVIEW

**Review by sobisht — CHANGES_REQUESTED** [review]
You don't have to add the node modules.

**Comment by nyteksf** [comment]
Thanks for giving my PR a look, @sobisht. I have since implemented the changes you'd pointed out regarding "node_modules/" folders. Please let me know if anything else needs changing, and I'll get to it. :)

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,10 @@
 Thumbs.db
 
+# ignore more deeply nested node_modules/ folders
```
Comment (sobisht) [inline .gitignore]: You should not add this file.

File: package.json [hunk 2]
```diff
@@ -24,6 +24,7 @@
         "lodash": "4.17.4",
         "npm": "3.10.10",
         "opn": "4.0.2",
+        "precommit": "^1.2.2",
```
Comment (sobisht) [inline package.json]: Is there any particular reason, this node module is added.

File: src/config.json [hunk 3]
```diff
@@ -99,4 +99,4 @@
             "url": "https://github.com/adobe/brackets/blob/master/LICENSE"
         }
     ]
-}
\ No newline at end of file
+}
```
Comment (sobisht) [inline src/config.json]: You don't have to add this file also.

File: src/editor/Editor.js [hunk 4]
```diff
@@ -89,6 +89,7 @@ define(function (require, exports, module) {
         LINEWISE_COPY_CUT   = "lineWiseCopyCut",
         SCROLL_PAST_END     = "scrollPastEnd",
         SHOW_CURSOR_SELECT  = "showCursorWhenSelecting",
+        SHOW_FULL_FILEPATH  = "showFullFilePathInTopBar",
```
Comment (sobisht) [inline src/editor/Editor.js]: Can you add this preference in DocumentCommandHandler.js only, since this preference is not used in Editor.js.
Also changing the preference name to showFullFilePath would be more generic compared to showFullFilePathInTopBar.

File: src/document/DocumentCommandHandlers.js [hunk 5]
```diff
@@ -159,8 +159,11 @@ define(function (require, exports, module) {
             windowTitle         = brackets.config.app_title,
             currentlyViewedFile = MainViewManager.getCurrentlyViewedFile(MainViewManager.ACTIVE_PANE),
             currentlyViewedPath = currentlyViewedFile.fullPath,
-            readOnlyString      = currentlyViewedFile.readOnly ? "[Read Only] - " : "";
-
+            readOnlyString      = currentlyViewedFile.readOnly ? "[Read Only] - " : "",
+            filepathDisplayPref = PreferencesManager.get('showFullPathToFile');
```
Comment (sobisht) [inline src/document/DocumentCommandHandlers.js]: Currently showFullPathToFile does not match with the name defined for preference 'showFullFilePathInTopBar'.

File: src/document/DocumentCommandHandlers.js [hunk 6]
```diff
@@ -159,8 +159,11 @@ define(function (require, exports, module) {
...
+            // Toggles between display of full or truncated filepath in top bar based on setting preference
+            var pathToFile = filepathDisplayPref === "false" ? _currentTitlePath : currentlyViewedPath;
```
Comment (sobisht) [inline src/document/DocumentCommandHandlers.js]: Can you change this tertiary operator from explicitly checking for "false" to something like
`var pathToFile = filepathDisplayPref ? currentlyViewedPath : _currentTitlePath;`

Existing check might give incorrect result.

File: src/nls/root/strings.js [hunk 7]
```diff
@@ -737,6 +737,7 @@ define({
     "DESCRIPTION_CLOSE_TAGS_WHEN_OPENING"            : "Close when > of opening tag is typed",
     "DESCRIPTION_CLOSE_TAGS_WHEN_CLOSING"            : "Close when / of closing tag is typed",
     "DESCRIPTION_CLOSE_TAGS_INDENT_TAGS"             : "An array of tags that when opened has a blank line",
+    "DESCRIPTION_SHOW_FULL_FILEPATH"                 : "true to display full file path in top bar",
```
Comment (sobisht) [inline src/nls/root/strings.js]: nit: replace 'in' by  'on'

**Review by sobisht — COMMENTED** [review]
Thanks for the PR @nyteksf. I have added my comments regarding the PR.

File: .gitignore (hunk 1 above)
Comment (nyteksf) [reply .gitignore]: Okay. Removed file.

File: src/config.json (hunk 3 above)
Comment (nyteksf) [reply src/config.json]: Thanks. I've removed this one too.

File: src/nls/root/strings.js (hunk 7 above)
Comment (nyteksf) [reply src/nls/root/strings.js]: Sure thing. Done.

File: src/document/DocumentCommandHandlers.js (hunk 6 above)
Comment (nyteksf) [reply src/document/DocumentCommandHandlers.js]: Thanks again. I've now made this change too.

File: src/document/DocumentCommandHandlers.js (hunk 5 above)
Comment (nyteksf) [reply src/document/DocumentCommandHandlers.js]: Okay. And I've changed the name as per your commented suggestion above.

File: package.json (hunk 2 above)
Comment (nyteksf) [reply package.json]: It was added when I was debugging an error preventing me from making PRs, but is not needed, so I've removed it.

**Comment by nyteksf** [comment]
Thanks for the help, @sobisht. If you can take a look once again when you get the chance, I've implemented your changes.

File: src/document/DocumentCommandHandlers.js [hunk 8]
```diff
@@ -159,8 +168,11 @@ define(function (require, exports, module) {
             windowTitle         = brackets.config.app_title,
             currentlyViewedFile = MainViewManager.getCurrentlyViewedFile(MainViewManager.ACTIVE_PANE),
             currentlyViewedPath = currentlyViewedFile.fullPath,
-            readOnlyString      = currentlyViewedFile.readOnly ? "[Read Only] - " : "";
-
+            readOnlyString      = currentlyViewedFile.readOnly ? "[Read Only] - " : "",
+            filepathDisplayPref = PreferencesManager.get('showFullFilePath');
```
Comment (sobisht) [inline src/document/DocumentCommandHandlers.js]: Variable name should follow the naming convention. 'p' in path should be capital.

File: src/document/DocumentCommandHandlers.js [hunk 9]
```diff
@@ -159,8 +168,11 @@ define(function (require, exports, module) {
...
+            // Toggles between display of full or truncated filepath in top bar based on setting preference
+            var pathToFile = filepathDisplayPref ? _currentTitlePath : currentlyViewedPath;
```
Comment (sobisht) [inline src/document/DocumentCommandHandlers.js]: - On setting filepathDisplayPref  to true, pathToFile should be set to currentlyViewedPath not _currentTitlePath.
- You don't have to use an extra var keyword.
- Can you change the name of variable 'pathToFile' to something related to Title as was for previous name '_currentTitlePath'

File: src/editor/Editor.js [hunk 10]
```diff
@@ -99,7 +99,7 @@ define(function (require, exports, module) {
         USE_TAB_CHAR        = "useTabChar",
         WORD_WRAP           = "wordWrap",
         AUTO_HIDE_SEARCH    = "autoHideSearch",
-        INDENT_LINE_COMMENT   = "indentLineComment",
+        INDENT_LINE_COMMENT = "indentLineComment",
```
Comment (sobisht) [inline src/editor/Editor.js]: This change is not related to the this PR.

**Comment by nyteksf** [comment]
I believe I've implemented all of your changes, @sobisht. If you could please verify, I would be grateful. Thanks.

**Review by nyteksf — COMMENTED** [review]
Hello @sobisht. I just needed to ask if there were still some changes you see that I've yet to make. I'm just wondering if you're waiting on me as I at least think I've implemented that which you've requested. Thanks for your time in any case.

**Comment by sobisht** [comment]
There are some issues with PR you have raised. Files:
- .gitignore
- package.json
- src/config.json

should not be present in PR. You have explicitly deleted these files via PR. If we merge the PRs then it will delete these files from the repo.
You just have to not include these files when submitting a commit.

File: src/document/DocumentCommandHandlers.js [hunk 11]
```diff
@@ -158,11 +166,13 @@ define(function (require, exports, module) {
         var currentDoc          = DocumentManager.getCurrentDocument(),
             windowTitle         = brackets.config.app_title,
             currentlyViewedFile = MainViewManager.getCurrentlyViewedFile(MainViewManager.ACTIVE_PANE),
-            currentlyViewedPath = currentlyViewedFile && currentlyViewedFile.fullPath,
-            readOnlyString      = (currentlyViewedFile && currentlyViewedFile.readOnly) ? "[Read Only] - " : "";
```
Comment (sobisht) [inline src/document/DocumentCommandHandlers.js]: You removed the change after resolving the conflict

File: src/document/DocumentCommandHandlers.js [hunk 12]
```diff
@@ -199,7 +209,7 @@ define(function (require, exports, module) {
             var projectName = projectRoot.name;
             // Construct shell/browser window title, e.g. "• index.html (myProject) — Brackets"
             if (currentlyViewedPath) {
-                windowTitle = StringUtils.format(WINDOW_TITLE_STRING_DOC, readOnlyString + _currentTitlePath, projectName, brackets.config.app_title);
+                windowTitle = StringUtils.format(WINDOW_TITLE_STRING_DOC, readOnlyString + DocumentTitle, projectName, brackets.config.app_title);  // -nyteksf
```
Comment (sobisht) [inline src/document/DocumentCommandHandlers.js]: variable name much start with lowercase.
remove comment // -nyteksf

File: src/editor/Editor.js [hunk 13]
```diff
@@ -99,7 +99,6 @@ define(function (require, exports, module) {
         USE_TAB_CHAR        = "useTabChar",
         WORD_WRAP           = "wordWrap",
         AUTO_HIDE_SEARCH    = "autoHideSearch",
-        INDENT_LINE_COMMENT   = "indentLineComment",
```
Comment (sobisht) [inline src/editor/Editor.js]: This file should not be present in the commit. Exclude this file. Don't delete it in PR.

File: src/document/DocumentCommandHandlers.js (hunk 11 above)
Comment (nyteksf) [reply src/document/DocumentCommandHandlers.js]: You mean the change to the file header? Which change was removed if otherwise please? I'm having trouble spotting the difference. Sorry if this is a newb question.

**Comment by nyteksf** [comment]
@sobisht, sorry to bother you, but can you please advise me on how to fix this situation with my PR? And how do I exclude files without using 'git rm --cached [file]', please?
